### ISSUE TITLE
support for presto-jdbc to generate Kerberos token based on userid and password

### ISSUE THREAD

**Issue by ankuragarawal** [prestosql/presto#4882]
In any server environment when we try to access Presto via the jdbc driver then you cannot have keytabs for multiple users or rely on credential cache.
It will be helpful to generate the Kerberos token with the username and password so that Kerberos Authentication can be performed in Presto while maintaining the identity of the requesting/querying user and we do not have to use a single system account for all the connections.

Would you be accepting a PR for this? I can do the required changes.

**Comment by electrum** [comment]
I’m not sure I follow. What information or credentials does the process have that is making the JDBC requests to Presto? Would it be trusted to impersonate other users?

The JDBC driver should already support impersonation for Kerberos. If you set the `user` parameter, that will request impersonation of that user. If you have a system access control configured, it will need to allow impersonation: https://prestosql.io/docs/current/security/built-in-system-access-control.html#impersonation-rules

**Comment by ankuragarawal** [comment]
I agree, we can add impresonation rules in built in system access control in Presto though we need the Kerberos token to be supplied by the driver to perform Kerberos authentication in Presto.

In this example the process will get username and password from the end user and there will be no impresonation in the  driver. It is just that rather than using a keytab or Kerberos credential cache( that we currently use in the driver 
for generating the Kerberos token), we ask the user to supply the credentials (username and password) for generating the Kerberos token.
Using a keytab/credential cache is almost impossible in a process where multiple connection requests (for different users or accounts) are routed via the driver and we want to perform Kerberos authentication for all incoming requests in Presto.

So the idea here is to add support in jdbc driver to accept username and password as connection properties for generating Kerberos token in addition to current process of accepting keytab or credential cache path(if KerberosRemoteServiceName is also supplied as a connection property)

**Comment by tooptoop4** [comment]
did u solve this @ankuragarawal ?

**Comment by ankuragarawal** [comment]
hi @tooptoop4. I somehow was not getting any notifications. I can add the required changes. Please let me know. Thanks
